source/test/Config.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2023 Klemens D. Morgenstern
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include "Config.hpp"
+#include <mrdox/Error.hpp>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+#include <llvm/Support/YAMLParser.h>
+#include <llvm/Support/YAMLTraits.h>
+
+
+template<>
+struct llvm::yaml::MappingTraits<
+    clang::mrdox::TestConfig>
+{
+  static void mapping(IO &io,
+                      clang::mrdox::TestConfig& f)
+  {
+    io.mapOptional("cxxstd", f.cxxstd);
+    io.mapOptional("compile-flags", f.compile_flags);
+    io.mapOptional("should-fail", f.should_fail);
+    io.mapOptional("heuristics", f.heuristics);
+  }
+};
+
+namespace clang {
+namespace mrdox {
+
+llvm::Expected<std::vector<TestConfig>>
+            loadTestConfig(llvm::StringRef dir,
+                           llvm::StringRef file)
+{
+    namespace fs = llvm::sys::fs;
+    namespace path = llvm::sys::path;
+
+    llvm::SmallString<256> filePath = file;
+    path::replace_extension(filePath, "yml");
+
+    if (!fs::exists(filePath))
+    {
+        filePath = dir;
+        filePath += "/mrdox-test.yml";
+    }
+
+    if (fs::exists(filePath))
+    {
+        auto fileText = llvm::MemoryBuffer::getFile(filePath);
+        if(! fileText)
+            return makeError(fileText.getError().message(), " when loading file '", filePath, "' ");
```

Comment:
I believe you can just put `fileText` here instead of `fileText.getError().message()`. If not, we can add a `nice` overload for it.